monitoring/rid_qualifier/flight_data_generator.py

```
@@ -0,0 +1,481 @@
+
+    def generate_flight_speed_bearing(self, adjacent_points: List, delta_time_secs: int) -> List[float]:
+        ''' A method to generate flight speed, assume that the flight has to traverse two adjecent points in x number of seconds provided, calculating speed in meters / second. It also generates bearing between this and next point, this is used to populate the 'track' paramater in the Aircraft State JSON. '''
+
+        first_point = adjacent_points[0]
+        second_point = adjacent_points[1]
+        
+        fwd_azimuth, back_azimuth, adjacent_point_distance_mts = self.geod.inv(first_point.x, first_point.y, second_point.x, second_point.y)
+        
+        speed_mts_per_sec = (adjacent_point_distance_mts / delta_time_secs)
+        speed_mts_per_sec = float("{:.2f}".format(speed_mts_per_sec))
+        
+        if fwd_azimuth < 0:
+            fwd_azimuth = 360 + fwd_azimuth
+            
+        return [speed_mts_per_sec, fwd_azimuth]
+
+    def utm_converter(self, shapely_shape: shapely.geometry, inverse:bool=False) -> shapely.geometry.shape:
+        ''' A helper function to convert from lat / lon to UTM coordinates for buffering. tracks. This is the UTM projection (https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system), we use Zone 33T which encompasses Switzerland, this zone has to be set for each locale / city. Adapted from https://gis.stackexchange.com/questions/325926/buffering-geometry-with-points-in-wgs84-using-shapely '''
+        
+        proj = Proj(proj="utm", zone=self.utm_zone, ellps="WGS84", datum="WGS84")
+        
+        geo_interface = shapely_shape.__geo_interface__
+        point_or_polygon = geo_interface['type']
+        coordinates = geo_interface['coordinates']
+        if point_or_polygon == 'Polygon':
+            new_coordinates = [[proj(*point, inverse=inverse) for point in linring] for linring in coordinates]
+        elif point_or_polygon == 'Point':
```

We should add an explicit `else` block here to catch the (presumably unexpected) situation where `point_or_polygon` is neither of these options.  It would be fine to, for instance, just throw an error in this block (`raise RuntimeError('Unexpected geo_interface type: {}'.format(point_or_polygon))`).  We want to fail fast and clearly to avoid mysterious errors in unexpected situations.

---

Thanks for this, yes, correct, I can anticipate a situation where we will be dealing with `MultiPolygons` etc. , not in this module but others.